src/pages/index.js

```diff
@@ -49,10 +110,62 @@ constants.addNewCardButton.addEventListener("click", () => {
 });
 
 function handleImageSubmit(card) {
-  renderCard(card);
-  addNewCardModal.close();
-  constants.imageSubmitForm.reset();
-  addImageFormValidator.resetValidation();
+  addNewCardModal.setLoading(true, "Saving...");
+  api
+    .addCard(card)
+    .then(() => {
+      renderCard(card);
+    })
+    .then(() => {
```

Comment:
Needs correction: here you should use `finally` as you should stop the loading in any case